### PR TITLE
util-linux-libuuid: Add version 2.40.2 and use Meson

### PR DESCRIPTION
**Pull request by jwillikers**
The upstream util-linux project is moving towards Meson. This PR migrates the recipe from Autotools to Meson for version 2.40.2. Version 2.39 has cross-compilation issues when using Meson.

Blocked by upstream issue causing build failures on macOS:

- [ ] https://github.com/util-linux/util-linux/issues/2873

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — DISMISSED** [review]

File: recipes/util-linux-libuuid/all/conanfile.py [hunk 1]
```diff
@@ -63,48 +65,64 @@ def _minimum_compiler_version(self, compiler, build_type):
...
         if "x86" in self.settings.arch:
-            tc.extra_cflags.append("-mstackrealign")
+            tc.c_args.append("-mstackrealign")
         tc.generate()
+        pkg_config_deps = PkgConfigDeps(self)
+        pkg_config_deps.generate()
+        virtual_build_env = VirtualBuildEnv(self)
+        virtual_build_env.generate()
+        virtual_run_env = VirtualRunEnv(self)
+        virtual_run_env.generate()
```
Comment (franramirez688) [inline recipes/util-linux-libuuid/all/conanfile.py:103]: @jwillikers why do we need both virtual env?

File: recipes/util-linux-libuuid/all/conanfile.py [hunk 2]
```diff
@@ -63,48 +65,64 @@ def _minimum_compiler_version(self, compiler, build_type):
...
+        copy(self, "COPYING.BSD-3-Clause", os.path.join(self.source_folder, "Documentation", "licenses"), os.path.join(self.package_folder, "licenses"))
+        meson = Meson(self)
+        meson.install()
         rm(self, "*.la", os.path.join(self.package_folder, "lib"))
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         rmdir(self, os.path.join(self.package_folder, "bin"))
         rmdir(self, os.path.join(self.package_folder, "sbin"))
         rmdir(self, os.path.join(self.package_folder, "share"))
+        # util-linux always builds both the shared and static libraries of libuuid, so delete the one that isn't needed.
+        rm(self, "libuuid.a" if self.options.shared else "libuuid.so*", os.path.join(self.package_folder, "lib"))
```
Comment (franramirez688) [inline recipes/util-linux-libuuid/all/conanfile.py:125]: Should we remove `libuuid.dylib*` too?

File: recipes/util-linux-libuuid/all/conanfile.py (hunk 1 above)
Comment (jwillikers) [reply recipes/util-linux-libuuid/all/conanfile.py:103]: I had it there for the `self.requires("libgettext/0.22")`.

File: recipes/util-linux-libuuid/all/conanfile.py (hunk 2 above)
Comment (jwillikers) [reply recipes/util-linux-libuuid/all/conanfile.py:125]: Yup, I spaced that one. Thanks!

**Comment by jwillikers** [comment]
Using Meson to cross-compile version 2.39 results in this error:

```
libmount/lib_mount.a.p/src_hook_mount.c.o -MF libmount/lib_mount.a.p/src_hook_mount.c.o.d -o libmount/lib_mount.a.p/src_hook_mount.c.o -c ../src/libmount/src/hook_mount.c
../src/libmount/src/hook_mount.c: In function ‘hook_create_mount’:
../src/libmount/src/hook_mount.c:302:40: error: ‘struct statx’ has no member named ‘stx_mnt_id’; did you mean ‘stx_uid’?
  302 |                 cxt->fs->id = (int) st.stx_mnt_id;
      |                                        ^~~~~~~~~~
      |                                        stx_uid
```

I've decided to target Meson for the upcoming 2.40 version. The 2.40-rc2 version cross-compiles successfully for me.

**Comment by jwillikers** [comment]
Ran into an issue: util-linux/util-linux#2830

**Pull request closed without merging**